src/DGmethods/DGmodel.jl

```diff
@@ -6,126 +6,112 @@ struct DGModel{BL,G,NFND,NFD,GNF}
   gradnumflux::GNF
 end
 
-function (dg::DGModel)(dQdt, Q, param, t; increment=false)
+function (dg::DGModel)(dYdt, Y, param, t; increment=false)
   bl = dg.balancelaw
-  device = typeof(Q.Q) <: Array ? CPU() : CUDA()
+  DFloat = eltype(Y)
+  device = typeof(Y.Q) <: Array ? CPU() : CUDA()
 
   grid = dg.grid
-  topology = grid.topology
-
-  dim = dimensionality(grid)
-  N = polynomialorder(grid)
-  Nq = N + 1
-  Nqk = dim == 2 ? 1 : Nq
-  Nfp = Nq * Nqk
-  nrealelem = length(topology.realelems)
-
-  Qvisc = param.diff
-  auxstate = param.aux
-
-  DFloat = eltype(Q)
-  nviscstate = num_diffusive(bl, DFloat)
+  Ω    = grid.topology.realelems
+  nΩ   = length(Ω)
+  Nᵈ   = dimensionality(grid)
+  N    = polynomialorder(grid)
+  Nq   = N + 1
+  Nqk  = Nᵈ == 2 ? 1 : Nq
+  Nfp  = Nq * Nqk
+  Np   = dofs_per_element(grid)
+
+  σ  = param.diff
+  α  = param.aux
+  nσ = num_diffusive(bl, DFloat)
 
-  lgl_weights_vec = grid.ω
-  Dmat = grid.D
   vgeo = grid.vgeo
   sgeo = grid.sgeo
-  vmapM = grid.vmapM
-  vmapP = grid.vmapP
-  elemtobndy = grid.elemtobndy
-  polyorder = polynomialorder(dg.grid)
-
-  Np = dofs_per_element(grid)
-
-  if hasmethod(update_aux!, Tuple{typeof(dg), typeof(bl), typeof(Q),
-                                  typeof(auxstate), typeof(t)})
-    update_aux!(dg, bl, Q, auxstate, t)
+  ω    = grid.ω
+  D    = grid.D
+  ι⁻   = grid.vmapM
+  ι⁺   = grid.vmapP
+  ιᴮ   = grid.elemtobndy
+
+  if hasmethod(update_aux!, Tuple{typeof(dg), typeof(bl), typeof(Y), typeof(α),
+                                  typeof(t)})
+    update_aux!(dg, bl, Y, α, t)
   end
 
   ########################
   # Gradient Computation #
   ########################
-  MPIStateArrays.start_ghost_exchange!(Q)
-  MPIStateArrays.start_ghost_exchange!(auxstate)
-
-  if nviscstate > 0
+  MPIStateArrays.start_ghost_exchange!(Y)
+  MPIStateArrays.start_ghost_exchange!(α)
 
-    @launch(device, threads=(Nq, Nq, Nqk), blocks=nrealelem,
-            volumeviscterms!(bl, Val(dim), Val(polyorder), Q.Q, Qvisc.Q, auxstate.Q, vgeo, t, Dmat,
-                             topology.realelems))
+  if nσ > 0
+    @launch(device, threads=(Nq, Nq, Nqk), blocks=nΩ,
+            volumeviscterms!(bl, Val(Nᵈ), Val(N), Y.Q, σ.Q, α.Q, vgeo, t, D, Ω))
```

Comment:
While you are at it, might as well replace `visc` by `diff` or `diffusive`, which is more generic and captures more broadly what these are about.

Reply:
I agree here.

Reply:
went with 

`volumeviscterms!` --> `volume_diffusive_terms!`
`faceviscterms!` --> `face_diffusive_terms!`

I think in the DG kernels we might want to replace diffusive with gradient or something similar, as not all second order terms are diffusive. it will also be more consistent if we do introduce integral variables and need to compute those here are well. for the higher level interfaces of AtmosModel we can still use diffusive though